examples/shader/post_process_pass.rs

```diff
@@ -0,0 +1,363 @@
+                BindGroupLayoutEntry {
+                    binding: 0,
+                    visibility: ShaderStages::FRAGMENT,
+                    ty: BindingType::Texture {
+                        sample_type: TextureSampleType::Float { filterable: true },
+                        view_dimension: TextureViewDimension::D2,
+                        multisampled: false,
+                    },
+                    count: None,
+                },
+                // The sampler that will be used to sample the screen texture
+                BindGroupLayoutEntry {
+                    binding: 1,
+                    visibility: ShaderStages::FRAGMENT,
+                    ty: BindingType::Sampler(SamplerBindingType::Filtering),
+                    count: None,
+                },
+            ],
+        });
+
+        // We can create the sampler here since it won't change at runtime and doesn't depend on the view
+        let sampler = render_device.create_sampler(&SamplerDescriptor::default());
+
+        // Get the shader handle
+        let shader = world
+            .resource::<AssetServer>()
+            .load("shaders/post_process_pass.wgsl");
+
+        let pipeline_id =
+            world
+                .resource_mut::<PipelineCache>()
+                .queue_render_pipeline(RenderPipelineDescriptor {
+                    label: Some("post_process_pipeline".into()),
+                    layout: Some(vec![layout.clone()]),
+                    // This will setup a fullscreen triangle for the vertex state
+                    vertex: fullscreen_shader_vertex_state(),
+                    fragment: Some(FragmentState {
+                        shader,
+                        shader_defs: vec![],
+                        entry_point: "fragment".into(),
+                        targets: vec![Some(ColorTargetState {
+                            format: TextureFormat::bevy_default(),
+                            blend: None,
+                            write_mask: ColorWrites::ALL,
+                        })],
+                    }),
+                    primitive: PrimitiveState::default(),
+                    depth_stencil: None,
+                    multisample: MultisampleState::default(),
+                });
+
+        Self {
+            layout,
+            sampler,
+            pipeline_id,
+        }
+    }
+}
+
+#[derive(Component, Default, Clone)]
+struct PostProcessSettings {
+    intensity: f32,
+}
+
+// TODO explain extract stage
+fn extract_post_process_settings(
+    mut commands: Commands,
+    cameras_2d: Extract<Query<(Entity, &Camera, &PostProcessSettings), With<Camera3d>>>,
+) {
+    for (entity, camera, settings) in &cameras_2d {
+        if camera.is_active {
+            commands.get_or_spawn(entity).insert(PostProcessUniform {
+                intensity: settings.intensity,
+            });
+        }
+    }
+}
+
+impl ExtractComponent for PostProcessSettings {
```

Comment:
I think if you change the query to also query for `Camera`, filter to `With<Camera3d>`, and the out type to be `PostProcessUniform`, you can then add `ExtractComponentPlugin` to the app world and you won't need `extract_post_process_settings`.

Reply:
I don't have the full code yet, but I'm trying to also show all the render stages in this example.